lib/course_planner/notifications.ex

```diff
@@ -17,13 +17,25 @@ defmodule CoursePlanner.Notifications do
   def to(%Notification{} = notification, %User{} = user),
     do: %{notification | user: user}
 
+  def add_data(%Notification{} = notification, data \\ %{}),
+    do: %{notification | data: data}
+
   def wake_up(now \\ DateTime.utc_now) do
     executed_at = Settings.get_value("NOTIFICATION_JOB_EXECUTED_AT", now)
     if Timex.diff(now, executed_at, :days) >= 1 do
+      build_all_notifications(now)
       send_all_notifications(now)
     end
   end
 
+  def build_all_notifications(now \\ DateTime.utc_now) do
+    if Settings.get_value("ENABLE_NOTIFICATION", true) do
```

Comment:
we check this when we send notifications already, don't you want to create the notification in the db at all?

Reply:
ohh that's true I should create them regardless of the "ENABLE_NOTIFICATION"

Reply:
done